server/runtime/src/main/java/org/infinispan/server/configuration/DataSourcesConfigurationBuilder.java

```diff
@@ -1,28 +1,39 @@
 package org.infinispan.server.configuration;
 
+import java.util.HashSet;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
 import java.util.stream.Collectors;
 
 import org.infinispan.commons.configuration.Builder;
 import org.infinispan.commons.configuration.attributes.AttributeSet;
+import org.infinispan.server.Server;
 
 public class DataSourcesConfigurationBuilder implements Builder<DataSourcesConfiguration> {
 
    private final AttributeSet attributes;
    private final ServerConfigurationBuilder server;
 
    private Map<String, DataSourceConfigurationBuilder> dataSources = new LinkedHashMap<>(2);
+   private Set<String> jndiNames = new HashSet<>(2);
 
    DataSourcesConfigurationBuilder(ServerConfigurationBuilder server) {
       this.server = server;
       attributes = DataSourcesConfiguration.attributeDefinitionSet();
    }
 
    DataSourceConfigurationBuilder dataSource(String name, String jndiName) {
+      if (dataSources.containsKey(name)) {
+         throw Server.log.duplicateDataSource(name);
```

Comment:
Similar to JDBC [1] you could log the exception as `CONFIG` logger.

[1] https://github.com/infinispan/infinispan/blob/master/persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/logging/Log.java#L30

Reply:
or not... it seems to be using the `SERVER` category for this in other places...

Reply:
This is a server configuration exception. It's fine as it is